docker/build_support/find-oldest-branch-root.sh

```diff
@@ -0,0 +1,6 @@
+#!/bin/sh
+
+set -e
+
+git log --pretty='format:%h %an' | grep GithubAction | head -n 1 | cut -d' ' -f1
```

Comment:
So, to greatly reduce test and image building in CI, we want to check for source files that were changed in commits since the last "merge commit" handled by github actions.  This script gets us the sha of that starting point, which allows us in `find-updated-sources` to determine if any files relevant to a specific directory have changed since the last github branch merging.